letsdrill.rb

```diff
@@ -1,13 +1,42 @@
 def get_letter_grade(integer)
- 
-  #Put your code here!
-
+  case integer
+  when 90..100
+    return "A"
+  when 80..89
+    return "B"
+  when 70..79
+    return "C"
+  when 60..69
+    return "D"
+  when 0..60
+    return "F"
+  else
+    print "Illegal grade input, must be number between 0 and 100"
+  end
 end
 
 def shortest_string(array)
 
-  #Put your code here!
+  #Make sure the input is an array and not empty
+  if array.empty? or not array.is_a? Array
+    print "Illegal input, must provide a non empty array\n"
+    return
+  end
 
+  shortest = nil
+  #Make sure the array only contains Strings
+  array.each do |x|
+    if not x.is_a? String
+      print "Array must only contain Strings\n"
+      return
+    end
+    
+    #We have a String lets see if its shortest
+    if shortest.nil? or x.size < shortest.size
```

Comment:
Ruby provides a bunch of higher-level methods to simplify stuff like this. [`Enumerable#min_by`](http://ruby-doc.org/core-2.4.2/Enumerable.html#method-i-min_by) should be a big help here. Scan through all those methods on Enumerable (Hash and Array too, if you feel like it). I think you'll find some neat stuff.